test/models/returns/ClaimedCreditsDetailsSpec.scala

```diff
@@ -16,48 +16,44 @@
+import models.returns.CreditsClaimedDetails._
 import org.scalatest.prop.TableDrivenPropertyChecks._
 import org.scalatestplus.play.PlaySpec
 import org.scalatestplus.scalacheck.ScalaCheckPropertyChecks.forAll
-import pages.returns.credits.{ConvertedCreditsPage, ConvertedCreditsWeightPage, ExportedCreditsPage, ExportedCreditsWeightPage, WhatDoYouWantToDoPage}
+import pages.returns.credits.{OldConvertedCreditsPage, OldExportedCreditsPage, WhatDoYouWantToDoPage}
 import viewmodels.PrintLong
 
 class ClaimedCreditsDetailsSpec extends PlaySpec {
 
   val userAnswer = UserAnswers("123")
-    .set(ExportedCreditsPage, true).get
-    .set(ExportedCreditsWeightPage, 100L).get
-    .set(ConvertedCreditsPage, true).get
-    .set(ConvertedCreditsWeightPage, 200L).get
+    .set(OldExportedCreditsPage, CreditsAnswer(true, Some(100L))).get
+    .set(OldConvertedCreditsPage, CreditsAnswer(true, Some(200L))).get
     .set(WhatDoYouWantToDoPage, true).get
 
   "summaryList" should {
     val table = Table(
       ("description", "exported", "converted", "exportedWeight", "convertedWeight"),
-      ("populate both weights when both answers are yes", true, true, 100L, 200L),
-      ("remove converted weight when converted is no", true, false, 100L, 0L),
-      ("remove exported weight when exported is no", false, true, 0L, 200L),
-      ("remove both exported and converted weight", false, false, 0L, 0L)
+      ("populate both weights when both answers are yes", true, true, Some(100L), Some(200L)),
+      ("remove converted weight when converted is no", true, false, Some(100L), None),
+      ("remove exported weight when exported is no", false, true, None, Some(200L)),
+      ("remove both exported and converted weight", false, false, None, None)
     )
 
     forAll(table) {
       (description, exported, converted, exportedWeight, convertedWeight) =>
         s"$description" in {
           val newAns = userAnswer
-            .set(ExportedCreditsPage, exported).get
-            .set(ExportedCreditsWeightPage, exportedWeight).get
-            .set(ConvertedCreditsPage, converted).get
-            .set(ConvertedCreditsWeightPage, convertedWeight).get
+            .set(OldExportedCreditsPage, CreditsAnswer(exported, exportedWeight)).get
+            .set(OldConvertedCreditsPage, CreditsAnswer(converted, convertedWeight)).get
 
           val credits = CreditsClaimedDetails(newAns, CreditBalance(10, 4, 200, true))
 
           credits.summaryList mustBe Seq(
             CreditExportedAnswerPartialKey -> (if (exported) "site.yes" else "site.no"),
-            CreditExportedWeightPartialKey -> exportedWeight.asKg,
+            exportedWeight.fold("N/A" -> "N/A")(o => CreditExportedWeightPartialKey -> o.asKg),
             CreditConvertedAnswerPartialKey -> (if (converted) "site.yes" else "site.no"),
-            CreditConvertedWeightPartialKey -> convertedWeight.asKg,
+            convertedWeight.fold("N/A" -> "N/A")(o => CreditConvertedWeightPartialKey -> o.asKg),
```

Comment:
is it "N/A" here or the weight? is it worth just doing a .get to make the test more deterministic?

Reply:
to-do list 👍